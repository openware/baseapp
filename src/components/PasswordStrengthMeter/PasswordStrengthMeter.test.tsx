import { shallow } from 'enzyme';
import * as React from 'react';
import { PasswordStrengthMeter, PasswordStrengthMeterProps } from './';

const defaults: PasswordStrengthMeterProps = {
    currentPasswordEntropy: 0,
    minPasswordEntropy: 0,
    passwordExist: false,
    passwordErrorFirstSolved: false,
    passwordErrorSecondSolved: false,
    passwordErrorThirdSolved: false,
    passwordPopUp: false,
    translate: jest.fn(),
};

const setup = (props: Partial<PasswordStrengthMeterProps> = {}) =>
    shallow(<PasswordStrengthMeter {...{...defaults, ...props }} />);

describe('PasswordStrengthMeter component', () => {
    it('should render', () => {
        const wrapper = setup();
        expect(wrapper).toMatchSnapshot();
    });

    it('renders without crashing', () => {
        const wrapper = setup();
        expect(wrapper).toBeDefined();
    });
});
