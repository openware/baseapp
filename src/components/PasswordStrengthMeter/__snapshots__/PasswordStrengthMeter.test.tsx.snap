// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PasswordStrengthMeter component should render 1`] = `ShallowWrapper {}`;
